homeassistant/components/device_tracker/__init__.py

```diff
@@ -594,6 +648,104 @@ def mark_stale(self):
                         state.attributes[ATTR_LONGITUDE])
 
 
+class DeviceTrackerEntity(Entity):
+    """Represent a tracked device."""
+
+    @property
+    def battery(self):
+        """Return the battery level of the device."""
+        return None
+
+    @property
+    def gps_accuracy(self):
+        """Return the gps accuracy of the device."""
+        return DEFAULT_GPS_ACCURACY
+
+    @property
+    def is_connected(self):
+        """Return True if device is connected."""
+        return None
+
+    @property
+    def latitude(self):
+        """Return latitude value of the device."""
+        return None
+
+    @property
+    def location_name(self):
+        """Return a location name for the current location of the device."""
+        return None
+
+    @property
+    def longitude(self):
+        """Return longitude value of the device."""
+        return None
+
+    @property
+    def mac_address(self):
+        """Return the mac address of the device."""
+        return None
+
+    @property
+    def source_type(self):
+        """Return the source type, eg gps or router, of the device."""
+        raise NotImplementedError
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self.location_name:
+            return self.location_name
+
+        if self.latitude is not None and self.source_type == SOURCE_TYPE_GPS:
+            zone_state = async_active_zone(
+                self.hass, self.latitude, self.longitude, self.gps_accuracy)
+            if zone_state is None:
+                state = STATE_NOT_HOME
+            elif zone_state.entity_id == zone.ENTITY_ID_HOME:
+                state = STATE_HOME
+            else:
+                state = zone_state.name
+            return state
+
```

Comment:
Should we add logic to determine the Area if possible?

Reply:
I think we can add new features later. But I'm open to suggestions.

Reply:
Agree - but we have an opportunity to do breaking changes now if we wanted to do away with `location_name` :)